### ISSUE TITLE
New MDK Broke Minifier Function

### ISSUE THREAD

**Issue by nicknds** [malware-dev/MDK-SE#151]
Scenario:
Visual Studio 2017 and freshly installed Visual Studio 2019
Using MDK 1.2.15
Using known working project and newly made projects
Using code typed in Notepad++ and checked using the in-game editor, error free

Actions:
Paste code into VS (done many times before)
Right-click project in side-bar, select "Deploy MDK Script"

Intended Result:
Build (I think)
Minify code
File created/ modified with minified code
Receive notification that code finished minifying

Actual Result:
Build (I think)
VS restarts (crashes?)
No file created/ modified
No notification

I don't understand how this was misinterpreted. I show a little frustration and I somehow it's not enough information. Here, it's so basic a child could get the same information from this as my unedited post.

> I tried minifying my updated script, VS restarted and no script was produced. I started a new project and got the same result.

All the information needed was here.

> I uninstalled VS 2017 and installed VS 2019. I opened the project I just made and tried to minify, VS restarted and produced nothing.
I started a new project checking remove unused type definitions and minify options. Tried minifying, same result.
I started another project only checking minify option, same result.
I started another project and checked nothing, it deployed the script but I need it minified so useless result.

Attempt to upgrade and attempt to fix errors from old project.

> Updated MDK. Used project I know worked, minifier broken. Updated VS. Used project I know worked and new project with same settings, minifier still broken.

Summed up, dumbed down.

Add some frustration and it's not enough information.

I don't see how uploading a failing project will help you at all. I said the project worked before and I said I tried making new projects. The project I use is essentially brand new. I paste code into a blank project, minify code, erase code from project, and save so it's new when I next use it.

**Comment by malware-dev** [comment]
I would appreciate if you could give me a little more information than just "it's broken"... 😛

**Comment by malware-dev** [comment]
For example, upload a failing project. The minifier works fine for me, and the prerelease had a couple hundred or so downloads before I even made a full release of it, and I had no complaints about the minifier... so I'm afraid this is nothing obvious.

**Comment by kwilliams1987** [comment]
Standard tech support request "it doesn't work".

**Comment by nicknds** [comment]
There you go. I would've simplified it sooner but I was busy. Thanks for all you do.

**Comment by malware-dev** [comment]
@nicknds With all due respect; I was simply trying to respond in kind. I don't appreciate the attitude. I do this _for free_ in my spare time, and gain _nothing_ from it - other than more work.

Like I said, I'm not seeing the error. It works for me, and nobody else has reported a problem _so far._ If I can't reproduce the problem, I can't fix it. It's not about "misinterpretation". I saw what you wrote. But there's no information in your post I can use. Your "simplification" doesn't change that. It doesn't crash for me, neither with a new nor a complex project, so if you want it fixed, you'll have to _help_ find the cause.

If you don't think uploading a failed project will help, then another option would be to try to run it with logging enabled in Visual Studio. Those logs are awful, but _maybe_ we can gleam something from it.

https://docs.microsoft.com/en-us/visualstudio/ide/reference/log-devenv-exe?view=vs-2019

If I _can_ help, I _will_ help - but for now, the ball is in your court.

**Comment by nicknds** [comment]
You could've started with "I can't reproduce this problem. I need more information. Could you upload a failing project? It could contain a way for me to reproduce the bug. Also could you try running VS with logging enabled and include the log in your upload? *Insert link for logging guide here*".

When someone tells me they have problems I ignore the lack of information, bad spelling and grammar, and their attitude. I just try and work with them. Tell them what I need to help them and how they can get that to me. I assume everyone knows nothing but what they observe and have no idea how to help besides reporting observations. I knew nothing but my observation and had no idea how to help besides reporting my observation.

I was trying to report a problem and express frustration in way that made you laugh at my misfortune.

As far as I know VS projects contain the source code, settings, and compiled code. I assumed since mine was empty, it had no valuable information and you could reproduce my entire project by just making a new one. Had you told me otherwise, like "It could contain a way for me to reproduce the bug" or "It could contain a crash or build log", then I would've uploaded it.

It looks like you got the information you needed from someone else and fixed the problem already. Next time I'll just downgrade or wait until it's fixed and not report any issues.

I appreciate everything you do. I know you do it for free and gain nothing from it. Have a good day.

**Comment by malware-dev** [comment]
@nicknds Yes, I could. But I'm not running a company, and as such I don't treat you guys as my _customers._ I was just having what I thought was harmless fun. I'm sorry I phrased this in a way that made you think that I was poking fun at you, but I wasn't. I'm not an English native, so I'll get phrasing wrong sometimes. From my point of view I _was_ telling you that I couldn't reproduce by saying that it works for me and nobody else reported a problem. That _still_ applies, by the way, so if I did something that fixed the problem for you that's completely coincidental. Still great though.
